dot/network/block_announce.go

```diff
@@ -183,6 +200,10 @@ func (s *Service) validateBlockAnnounceHandshake(from peer.ID, hs Handshake) err
 		return errors.New("invalid handshake type")
 	}
 
+	if bhs.Roles != FullNode && bhs.Roles != LightClient && bhs.Roles != Validator {
+		return errInvalidRole
+	}
+
```

Comment:
@qdm12, @kishansagathiya I think there is a pattern in the role numbers `1, 2, 4`, which looks like when we raise 2 to the power of 0, 1, and 2 respectively. So instead of checking for `1, 2 or 4` separately you could do:

```go
type Role byte

const (
	FullNode Roles = 1 << iota // will be evaluated to 1
	LightClient Roles // will be evaluated to 2
	Validator Roles // will be evaluated to 4
)

...
// then in the getBlockAnnounceHandshake method
func (s *Service) getBlockAnnounceHandshake() (Handshake, error) {
  ...
  switch bhs.Roles {
    case 1, 2, 4:
    default:
      return errors.New("inexistent role")
    }
}
```